# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidence to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?


In [263]:
#import pandas
import pandas as pd

In [316]:
#import data from csv
abbdata = pd.read_csv("AB_NYC_2019.csv")
abbdata

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

In [402]:
#Which host was the busiest?

#Which host has the most listings?

host_with_most_listings = (abbdata.groupby('host_id').sum(True).sort_values('price', ascending = False)['price'].head())
single_top_host = top_hosts.index[0]

single_top_host

display(abbdata.groupby('calculated_host_listings_count').count().sort_values("calculated_host_listings_count", ascending=False))

#host 327 is the busiest as the host as the most listings 


Object `listings` not found.


id   name  host_id  host_name  \
calculated_host_listings_count                                     
327                               327    327      327        327   
232                               232    232      232        232   
121                               121    121      121        121   
103                               103    103      103        103   
96                                192    192      192        192   
91                                 91     91       91         91   
87                                 87     87       87         87   
65                                 65     65       65         65   
52                                104    104      104        104   
50                                 50     50       50         50   
49                                 98     98       98         98   
47                                 47     47       47         47   
43                                 43     43       43         43   
39                                 39     39       39         39   
37                                 37     37       37         37   
34                                 68     68       68         68   
33                                 99     99       99         99   
32                                 32     32       32         32   
31                                 62     62       62         62   
30                                 30     30       30         30   
29                                 29     29       29         29   
28                                 56     56       56         56   
27                                 27     27       27         27   
26                                 26     26       26         26   
25                                 50     50       50         50   
23                                 69     69       69         69   
21                                 21     21       21         21   
20                                 40     40       40         40   
19                                 19     19       19         19   
18                                 54     54       54         54   
17                                 68     68       68         68   
16                                 16     16       16         16   
15                                 75     75       75         75   
14                                 70     70       70         70   
13                                130    130      130        130   
12                                180    180      180        180   
11                                110    110      110        110   
10                                210    210      210        210   
9                                 234    234      234        234   
8                                 416    416      416        416   
7                                 399    399      399        399   
6                                 570    570      570        570   
5                                 845    844      845        845   
4                                1440   1440     1440       1440   
3                                2853   2853     2853       2850   
2                                6658   6657     6658       6656   
1                               32303  32289    32303      32287   

                                neighbourhood_group  neighbourhood  latitude  \
calculated_host_listings_count                                                 
327                                             327            327       327   
232                                             232            232       232   
121                                             121            121       121   
103                                             103            103       103   
96                                              192            192       192   
91                                               91             91        91   
87                                               87           

In [374]:
#2 How many neighborhood groups are available and which shows up the most?

print(abbdata.groupby('neighbourhood_group').size())
print(abbdata.groupby('neighbourhood_group').size().count())

#There are 5 neighborhoods and Manhattan  has the most rentals on AirBnB




neighbourhood_group
Bronx             1091
Brooklyn         20104
Manhattan        21661
Queens            5666
Staten Island      373
dtype: int64
5


In [57]:
#Are private rooms the most popular in manhattan?



manhattan_list = abbdata[abbdata['neighbourhood_group'] == "Manhattan"]

display(manhattan_list.groupby('room_type').size())

manhattan_list_proptypecount = (manhattan_list.groupby('room_type').size())

manhattan_list_proptypecount


f'The most popiular type of AirBnb properties are {manhattan_list_proptypecount.index[0]}'



room_type
Entire home/apt    13199
Private room        7982
Shared room          480
dtype: int64

'The popularity of AirBnb properties was Entire home/apt'

In [385]:
# Which hosts are the busiest and based on their reviews?

display(abbdata.groupby('host_id').count()['reviews_per_month'])

#This is not a good way to measure host business we do not have any indication of the quality of the reviews. We chose to use reviews/month to indicate comparison over a similar time span since the longer the listing has been available and the number of guests will significantly impact number of reviews. 



host_id
2438         1
2571         1
2787         6
2845         2
2868         1
            ..
274273284    0
274298453    0
274307600    0
274311461    0
274321313    0
Name: reviews_per_month, Length: 37457, dtype: int64

In [58]:
#Which neighorhood group has the highest average price?

display(abbdata.groupby('neighbourhood_group').mean(True)['price'])

#the highest average price is Manhattan


neighbourhood_group
Bronx             87.496792
Brooklyn         124.383207
Manhattan        196.875814
Queens            99.517649
Staten Island    114.812332
Name: price, dtype: float64

In [388]:
# Which neighborhood group has the highest total price?

display(abbdata.groupby('neighbourhood_group').sum(True)['price'])

#the highest total price is Manhattan. This data is influenced by both average unit prices and number of units available. 



neighbourhood_group
Bronx              95459
Brooklyn         2500600
Manhattan        4264527
Queens            563867
Staten Island      42825
Name: price, dtype: int64

In [73]:
#Which top 5 hosts have the highest total price?

display(abbdata.groupby('host_id').sum(True).sort_values('price', ascending = False)['price'].head())

#The hosts listed below are of make up the highest total price. 

host_id
219517861    82795
107434423    70331
156158778    37097
205031545    35294
30283594     33581
Name: price, dtype: int64

In [77]:
# Who currently has no (zero) availability with a review count of 100 or more?
                                                         


#mask for review count
#mask for zero availability

reviews_mask = abbdata["number_of_reviews"] >= 100


availability_mask = abbdata ["availability_365"] == 0


display(abbdata[reviews_mask & availability_mask].shape)

abbdata[reviews_mask & availability_mask]

#the list below contains hosts with 0 availability
#zero availibility with 100+ reviews is not a true measure activity because the number of years listed influences the number of reviews. Additionally, units have zero availability when the host does not list the property as well as when it is fully occupied. 




(162, 16)

id                                       name    host_id  \
8          5203         Cozy Clean Guest Room - Family Apt       7490   
94        20913         Charming 1 bed GR8 WBurg LOCATION!      79402   
132       30031           NYC artists’ loft with roof deck     129352   
174       44221             Financial District Luxury Loft     193722   
180       45556      Fort Greene, Brooklyn: Center Bedroom      67778   
...         ...                                        ...        ...   
29581  22705516         The Quietest Block in Manhattan :)  127740507   
30461  23574142                          queens get away!!  176185168   
31250  24267706  entire sunshine of the spotless mind room   21074914   
32670  25719044                COZY Room for Female Guests   40119874   
35014  27759146     Cozy corner near Empire State Building  209549523   

        host_name neighbourhood_group              neighbourhood  latitude  \
8       MaryEllen           Manhattan            Upper West Side  40.80178   
94     Christiana            Brooklyn               Williamsburg  40.70984   
132           Sol            Brooklyn                 Greenpoint  40.73494   
174         Coral           Manhattan         Financial District  40.70666   
180          Doug            Brooklyn                Fort Greene  40.68863   
...           ...                 ...                        ...       ...   
29581    Kathleen           Manhattan                     Harlem  40.83102   
30461       Janet              Queens                  Laurelton  40.68209   
31250      Albert            Brooklyn         Bedford-Stuyvesant  40.68234   
32670    Stephany            Brooklyn  Prospect-Lefferts Gardens  40.66242   
35014     Mariluz           Manhattan                    Midtown  40.74858   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
8      -73.96723     Private room     79               2                118   
94     -73.95775  Entire home/apt    100               5                168   
132    -73.95030     Private room     50               3                193   
174    -74.01374  Entire home/apt    196               3                114   
180    -73.97691     Private room     65               2                206   
...          ...              ...    ...             ...                ...   
29581  -73.94181     Private room     65               2                103   
30461  -73.73662     Private room     65               1                119   
31250  -73.91318     Private room     49               1                102   
32670  -73.94417     Private room     48               1                131   
35014  -73.98341      Shared room     62               1                112   

      last_review  reviews_per_month  calculated_host_listings_count  \
8      2017-07-21               0.99                               1   
94     2018-07-22               1.57                               1   
132    2019-05-20               1.86                               1   
174    2019-06-20               1.06                               1   
180    2019-06-30               1.92                               2   
...           ...                ...                             ...   
29581  2019-07-07               5.89                               2   
30461  2018-12-24               7.79                               1   
31250  2019-07-05               6.73                               3   
32670  2019-05-31               9.97                               2   
35014  2019-06-13              10.77                               3   

       availability_365  
8                     0  
94                    0  
132                   0  
174                   0  
180                   0  
...                 ...  
29581                 0  
30461                 0  
31250                 0  
32670                 0  
35014                 0  

[162 rows x 16 columns]

In [392]:
# What host has the highest total of prices and where are they located?

top_hosts = (abbdata.groupby('host_id').sum(True).sort_values('price', ascending = False)['price'])
single_top_host = top_hosts.index[0]


display(top_hosts)

display(single_top_host)

#Host 219517861 has the highest total prices 













host_id
219517861    82795
107434423    70331
156158778    37097
205031545    35294
30283594     33581
             ...  
205820814       10
180661875       10
11967922        10
10132166         0
13709292         0
Name: price, Length: 37457, dtype: int64

219517861

In [398]:
# When did Danielle from Queens last receive a review?
import pandas as pd

import numpy as np


danielle_data_location = abbdata[(abbdata.host_name == "Danielle") & (abbdata.neighbourhood_group == "Queens")]
print(danielle_data_location.sort_values("last_review", ascending=False))

#Daniel from Quennes last recieved a review on July 8 2019


             id                                               name    host_id  \
22469  18173787  Cute Tiny Room Family Home by LGA NO CLEANING FEE   26432133   
21517  17222454   Sun Room Family Home LGA Airport NO CLEANING FEE   26432133   
20403  16276632  Cozy Room Family Home LGA Airport NO CLEANING FEE   26432133   
22068  17754072                Bed in Family Home Near LGA Airport   26432133   
7086    5115372  Comfy Room Family Home LGA Airport NO CLEANING...   26432133   
33861  26814763  One bedroom with full bed / 1 stop from Manhattan  201647469   
27021  21386105  Quiet & clean 1br haven with balcony near the ...  154256662   
16349  13151075                    ASTORIA APARTMENT OUTDOOR SPACE   18051286   

      host_name neighbourhood_group     neighbourhood  latitude  longitude  \
22469  Danielle              Queens     East Elmhurst  40.76380  -73.87238   
21517  Danielle              Queens     East Elmhurst  40.76367  -73.87088   
20403  Danielle              Queens 

## Further Questions

1. Which host has the most listings?





In [400]:
host_with_most_listings = (abbdata.groupby('host_id').sum(True).sort_values('price', ascending = False)['price'].head())
single_top_host = top_hosts.index[0]

single_top_host

display(abbdata.groupby('calculated_host_listings_count').count().sort_values("calculated_host_listings_count", ascending=False))


display(host_with_most_listings)
display(single_top_host)
#

f'host {single_top_host} has the most listings'

id   name  host_id  host_name  \
calculated_host_listings_count                                     
327                               327    327      327        327   
232                               232    232      232        232   
121                               121    121      121        121   
103                               103    103      103        103   
96                                192    192      192        192   
91                                 91     91       91         91   
87                                 87     87       87         87   
65                                 65     65       65         65   
52                                104    104      104        104   
50                                 50     50       50         50   
49                                 98     98       98         98   
47                                 47     47       47         47   
43                                 43     43       43         43   
39                                 39     39       39         39   
37                                 37     37       37         37   
34                                 68     68       68         68   
33                                 99     99       99         99   
32                                 32     32       32         32   
31                                 62     62       62         62   
30                                 30     30       30         30   
29                                 29     29       29         29   
28                                 56     56       56         56   
27                                 27     27       27         27   
26                                 26     26       26         26   
25                                 50     50       50         50   
23                                 69     69       69         69   
21                                 21     21       21         21   
20                                 40     40       40         40   
19                                 19     19       19         19   
18                                 54     54       54         54   
17                                 68     68       68         68   
16                                 16     16       16         16   
15                                 75     75       75         75   
14                                 70     70       70         70   
13                                130    130      130        130   
12                                180    180      180        180   
11                                110    110      110        110   
10                                210    210      210        210   
9                                 234    234      234        234   
8                                 416    416      416        416   
7                                 399    399      399        399   
6                                 570    570      570        570   
5                                 845    844      845        845   
4                                1440   1440     1440       1440   
3                                2853   2853     2853       2850   
2                                6658   6657     6658       6656   
1                               32303  32289    32303      32287   

                                neighbourhood_group  neighbourhood  latitude  \
calculated_host_listings_count                                                 
327                                             327            327       327   
232                                             232            232       232   
121                                             121            121       121   
103                                             103            103       103   
96                                              192            192       192   
91                                               91             91        91   
87                                               87           

host_id
219517861    82795
107434423    70331
156158778    37097
205031545    35294
30283594     33581
Name: price, dtype: int64

219517861

'host 219517861 has the most listings'

2. How many listings have completely open availability?

In [324]:

availibility_list = abbdata[abbdata['availability_365'] == 365]

availibility_list.count()

#1295 is the listing with completely open availability 

id                                1295
name                              1294
host_id                           1295
host_name                         1295
neighbourhood_group               1295
neighbourhood                     1295
latitude                          1295
longitude                         1295
room_type                         1295
price                             1295
minimum_nights                    1295
number_of_reviews                 1295
last_review                        841
reviews_per_month                  841
calculated_host_listings_count    1295
availability_365                  1295
dtype: int64

3. What room_types have the highest review numbers?

In [401]:
most_reviewed_room_type = (abbdata.groupby('room_type').sum(True).sort_values('number_of_reviews', ascending = False)['price'].head())
single_most_reviewed = most_reviewed_room_type.index[0]

single_most_reviewed

f'{single_most_reviewed} "have the highest number of reviews'





'Entire home/apt "have the highest number of reviews'

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

1. Which hosts are the busiest and why?
    #host 327 is the busiest as the host as the most listings 
    
2. How many neighborhood groups are available and which shows up the most?
    5, Manhattan

3. Are private rooms the most popular in manhattan?
    No, Entire units

4. Which hosts are the busiest and based on their reviews?
    2438         1
    2571         1
    2787         6
    2845         2
    2868         1
    

5. Which neighorhood group has the highest average price?
    The highest average price is Manhattan
    
6. Which neighborhood group has the highest total price?
    The highest neighborhood group price is Manhattan

7. Which top 5 hosts have the highest total price?
    219517861    82795
    107434423    70331
    156158778    37097
    205031545    35294
    30283594 

8. Who currently has no (zero) availability with a review count of 100 or more?
    There are 162 hosts with zero availability and more than 100 reviews
    
9. What host has the highest total of prices and where are they located?
    219517861    

10. When did Danielle from Queens last receive a review?
    July 8 2019


Also, if you uncovered some more details that were not asked above, please describe them here.

-- Add your conclusion --

Manhattan is a popular place for AirBnBs but additional research should be put into whether Staten Island is ripe for investment due to low competition. This would include understanding how often listings are available and what occupancy rates are. 

Availability does not indicate bookings because a unit can be unanavailable when it is booked or when the host has not listed it. 

Number of reviews requires additonal information to understand how high the reviews are. Additionally, time on AirBnB and number of listings affect the humber of total reviews 

Total prices is not a good indicator of affordability since it is influenced both by number of units in a given location and price of those units.


